File: src/ModuleSampleForVB/AuthenticationModule/AuthenticationModule1.vb
Imports System.ComponentModel.Composition
Imports System.Threading
Imports Storex

<Export(GetType(IAuthenticationModule))>
<ExportMetadata(NameOf(IAuthenticationModuleMetadata.Id), "FCB29577-7E5B-4B0C-A514-B8E636AAF13D")>
<ExportMetadata(NameOf(IAuthenticationModuleMetadata.DisplayName), "簡易認証 (ID入力)")>
<ExportMetadata(NameOf(IAuthenticationModuleMetadata.Description), "CSVファイル (ID と 氏名) を使用します")>
Public Class AuthenticationModule1
    Implements IAuthenticationModule

    Public ReadOnly Property IsConfiguable As Boolean Implements IAuthenticationModule.IsConfiguable
        Get
            IsConfiguable = True
        End Get
    End Property

    Public Function ConfigureAsync() As Task Implements IAuthenticationModule.ConfigureAsync

        AuthenticationModuleHelper.PickUsersFileAndSaveToSetting()

        Return Task.CompletedTask
    End Function

    Public Async Function AuthenticateAsync() As Task(Of IUserInfo) Implements IAuthenticationModule.AuthenticateAsync

        Return Await AuthenticateAsync(cancellationToken:=CancellationToken.None)
    End Function

    Public Function AuthenticateAsync(cancellationToken As CancellationToken) As Task(Of IUserInfo) Implements IAuthenticationModule.AuthenticateAsync

        Dim User As IUserInfo = AuthenticationForm1.Authenticate

        Return Task.FromResult(User)
    End Function

    Public Sub Dispose() Implements IDisposable.Dispose
    End Sub
End Class

File: src/AuthenticationModules.CS/AuthenticationModule1.cs
using System.Threading;
using System.Threading.Tasks;
using Storex;

namespace AuthenticationModules
{
    [AuthenticationModuleExport("FCB29577-7E5B-4B0C-A514-B8E636AAF13D", "簡易認証 (ID入力)", Description = "CSVファイル (ID と 氏名) を使用します")]
    public class AuthenticationModule1 : IAuthenticationModule
    {
        public bool IsConfiguable => true;

        public Task ConfigureAsync(CancellationToken cancellationToken)
        {
            AuthenticationModuleHelper.PickUsersFileAndSaveToSetting();

            return Task.CompletedTask;
        }

        public Task<IUser> AuthenticateAsync(CancellationToken cancellationToken)
        {
            IUser user = AuthenticationForm1.Authenticate();

            return Task.FromResult(user);
        }

        public void Dispose()
        {
        }
    }
}